internal/metamorph/bcnet/metamorph_p2p/message_handler.go

```diff
@@ -0,0 +1,180 @@
+package metamorph_p2p
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/bitcoin-sv/arc/internal/metamorph/metamorph_api"
+	"github.com/bitcoin-sv/arc/internal/metamorph/store"
+	"github.com/bitcoin-sv/arc/internal/p2p"
+	"github.com/libsv/go-p2p/bsvutil"
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+	"github.com/libsv/go-p2p/wire"
+)
+
+var ErrTxRejectedByPeer = errors.New("transaction rejected by peer")
+
+type PeerTxMessage struct {
+	Start        time.Time
+	Hash         *chainhash.Hash
+	Status       metamorph_api.Status
+	Peer         string
+	Err          error
+	CompetingTxs []string
+}
+
+var _ p2p.MessageHandlerI = (*MsgHandler)(nil)
+
+type MsgHandler struct {
+	logger    *slog.Logger
+	store     store.MetamorphStore
+	messageCh chan<- *PeerTxMessage
+}
+
+func NewMsgHandler(l *slog.Logger, s store.MetamorphStore, messageCh chan<- *PeerTxMessage) *MsgHandler {
+	ph := &MsgHandler{
+		logger:    l,
+		store:     s,
+		messageCh: messageCh,
+	}
+
+	return ph
+}
+
+// should be fire & forget
```

Comment:
Change comment to 

// OnReceive handles incoming messages depending on command type

Reply:
Done